helm-charts/basehub/values.yaml

```
@@ -400,7 +404,77 @@ jupyterhub:
                   - value: "/rstudio"
                     title: RStudio
                     description: An IDE For R, created by the RStudio company
-
+    initContainers:
+      - name: templates-clone
+        image: alpine/git
+        args:
+          - clone
+          - --
+          - $(HUB_TEMPLATES_REPO)
+          - /srv/repo
+        env:
+          - name: HUB_TEMPLATES_REPO
+            valueFrom:
+              configMapKeyRef:
+                name: hub-custom-templates-config
+                key: HUB_TEMPLATES_REPO
+        securityContext:
+          runAsUser: 1000
+          runAsGroup: 1000
+          allowPrivilegeEscalation: False
+          readOnlyRootFilesystem: True
+        volumeMounts:
+          - name: custom-templates
+            mountPath: /srv/repo
+      - name: templates-ownership-fix
+        image: alpine/git
+        command:
+          - /bin/sh
+        args:
+          - -c
+          - ls -lhd /srv/repo && chown 1000:1000 /srv/repo && ls -lhd /srv/repo
+        securityContext:
+          runAsUser: 0
+        volumeMounts:
+          - name: custom-templates
+            mountPath: /srv/repo
+    extraContainers:
+      - name: templates-sync
+        image: alpine/git
+        workingDir: /srv/repo
+        command:
+          - /bin/sh
+        args:
+          - -c
+          - |
+            ls -lhd /srv/repo;
+            while true; do git fetch origin;
+            git reset --hard origin/$(HUB_TEMPLATES_REPO_BRANCH);
```

I think the syntax for referencing an env variable to use here, is valid if you were in the k8s `env` section defining another env var based on another env var.

This is the container command, inside a `sh` based shell though, so I think the syntax should be `$SOMETHING` or `${SOMETHING}` instead.

```suggestion
            git reset --hard origin/${HUB_TEMPLATES_REPO_BRANCH};
```

---

Ah ignore me, this will work using k8s env var expansion and doesn't have to rely on the shell env var expansion.

---

Btw @consideRatio, I've tested this on staging hub and it appeared to do the correct thing, so at least there shouldn't be syntax issues. (I'm doing it again now to double check 😅 )